Modules/mathmodule.c

```diff
@@ -3017,103 +3017,105 @@ Raises ValueError if the arguments are negative or if k > n.
 
 static PyObject *
 math_comb_impl(PyObject *module, PyObject *n, PyObject *k)
-/*[clinic end generated code: output=bd2cec8d854f3493 input=565f340f98efb5b5]*/
+/*[clinic end generated code: output=bd2cec8d854f3493 input=b2160da8fe59df60]*/
 {
-    PyObject *val = NULL,
-        *temp_obj1 = NULL,
-        *temp_obj2 = NULL,
-        *dump_var = NULL;
+    PyObject *result = NULL, *factor = NULL, *temp;
     int overflow, cmp;
-    long long i, terms;
+    long i, factors;
+
+    n = PyNumber_Index(n);
+    if (n == NULL)
+        return NULL;
+    k = PyNumber_Index(k);
+    if (k == NULL) {
+        Py_DECREF(n);
+        return NULL;
+    }
 
     cmp = PyObject_RichCompareBool(n, k, Py_LT);
     if (cmp < 0) {
-        goto fail_comb;
+        goto error;
     }
     else if (cmp > 0) {
-        PyErr_Format(PyExc_ValueError,
-                     "n must be an integer greater than or equal to k");
-        goto fail_comb;
+        PyErr_SetString(PyExc_ValueError,
+                        "n must be an integer greater than or equal to k");
```

Comment:
While this error message is technically accurate, it reads oddly to me: I think of the `k <= n` condition as a constraint on `k` rather than `n`, so I'd write that as `"k must be an integer less than or equal to n"`.

@serhiy-storchaka What do you think? Is this just me that this reads oddly to?